### PR TITLE
Add other workflows and start to update documentation.

### PR DESCRIPTION
**Pull request by iamsophiesk**
Marking this as ready for review. Once this is merged we may still want/need to make some adjustments as we migrate sites across to the GAIN Github/Github Actions.

* Added extra workflow files.
  - Renamed them all numerically to provide some order (as they need to be in the root directory):
    * `00`: Helper workflows.
    * `01`: Build tasks.
    * `02`: Test tasks.
    * `03`: Deploy tasks.
    * `04`: none yet, but see these being scan tasks. 
  - Renamed the existing `composer.yml` file to `01-build-composer.yml` to match the others.
* Added a couple of actions that we use around the place as well.
* Added README file. I hope you all like readmes. I like readmes. This is a hell of a readme.
* Added some example workflows based on the projects we have in our old GitHub repo.
* Added some new workflows that I haven't tested yet, based on some things we have in projects I'm on... cannot guarantee they'll work.
* Added a `MIGRATION.md` file that I'm hoping to populate as a cheat sheet for using these files. The Access one will be easy - but I might lean on others to populate other agency sections.
* Updated all (hopefully!) secret/input variable names, and updated a ton of documentation, including populating the `MIGRATION.md` file for Access -> GAIN migrations.
* Added a `CODEOWNERS` file and marked all tech leads as owners.

These workflow updates have been tested on the following PRs:
* Adder Technology:
  - Access Github, already using Access workflows
  - PR: https://github.com/accessdigital/adder/pull/664
  - Successful actions run: https://github.com/accessdigital/adder/actions/runs/24453716416?pr=664 
  - Deployed successfully to a multidev site build: https://t-t-650-upd-adder-technology.pantheonsite.io/
* IAB:
  - Catch Bitbucket, migrating into GAIN Github
  - PR: https://github.com/thisisgain/iab-website/pull/1
  - Successful actions run: https://github.com/thisisgain/iab-website/actions/runs/24566805189?pr=1 
  - Deployed successfully to the dev2 environment: https://dev2.iabuk.com/
* SWBS Marketing:
  - Access Github, already using Access workflows
  - PR: https://github.com/accessdigital/SWBS-marketing-EE/pull/148 
  - Successful test actions run: https://github.com/accessdigital/SWBS-marketing-EE/actions/runs/24569203431?pr=148
  - Successful deploy actions run: https://github.com/accessdigital/SWBS-marketing-EE/actions/runs/24571709835

Still to do, but don't block review/merge: 
* Add more example workflows - in particular, I want to add "build a Node app and combine with a CMS build for deployment".
* More tests - none of the sites I'm working on currently use a manual workflow dispatch - would be nice to try that out. 
* Add a `CONTRIBUTING.md` file, as well as other metafiles (eg PR template, etc), though these can be done at a later date.

### CODE REVIEW

File: .github/examples/test-postman.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# Cancel out of date jobs.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  # Download dependencies to build a project using a Composer lockfile.
+  composer-build:
+    name: 'Composer: build project.'
+    uses: accessdigital/.github/.github/workflows/composer.yml@main
```
Comment (BenGarry-GAIN) [inline .github/examples/test-postman.yml]: Assume these accessdigital references need to point to thisisgain?

Comment (iamsophiesk) [reply .github/examples/test-postman.yml]: Yes, see the "to do" list in the main PR description, this is still a draft :)

**Comment by iamsophiesk** [comment]
Howdy @thisisgain/gain-experience-tech-leads - this is ready to be reviewed :) 

I've tested these on a few different client sites: two that already used the Access GHA but deployed differently, and a Catch site that needs to be migrated from Bitbucket into Github. I've linked the PRs and successful actions as appropriate.

There are still some more workflows to try out/make sure they're working, but I've come to the end of the projects I'm currently active on. I'm happy to support with any other migrations, or if anybody else wants to try these out, that'd be dandy. 

I'm not sure which sites are currently using the `composer` action that I've renamed/replaced - it would be good to confirm that the `01-composer-build` action works as a direct replacement.

**Review by mangyfox — APPROVED** [review]
